crates/ark/src/lsp/symbols.rs

```diff
@@ -95,6 +95,19 @@ pub fn symbols(params: &WorkspaceSymbolParams) -> anyhow::Result<Vec<SymbolInfor
                     container_name: None,
                 });
             },
+            IndexEntryData::Variable { name } => {
+                info.push(SymbolInformation {
+                    name: name.clone(),
+                    kind: SymbolKind::VARIABLE,
+                    location: Location {
+                        uri: Url::from_file_path(path).unwrap(),
+                        range: entry.range,
+                    },
+                    tags: None,
+                    deprecated: None,
+                    container_name: None,
+                });
+            },
         };
     });
 
```

Comment:
It's somewhat interesting to note that for _document symbols_, i.e. with `@`, we were already tracking all variables due to how `document_symbols()` and eventually `index_assignment()` works.

It even already worked with `method(` calls. We don't check that the LHS is an identifier, we just accept anything. That does lead to some weirdness but its probably ok.

<img width="620" alt="Screenshot 2024-11-05 at 8 58 59 AM" src="https://github.com/user-attachments/assets/92b2e648-53e0-4b38-8219-e80c5706c907">

So it was just the following two places where we weren't indexing variables but we do now:
- Workspace level symbols (so `#name` works across files, and so you can jump to defn)
- Workspace level static analysis completions (even though we previously did get a completion item already due to runtime completions once the user loaded the package)

Reply:
oh I hadn't caught up on that, good catch